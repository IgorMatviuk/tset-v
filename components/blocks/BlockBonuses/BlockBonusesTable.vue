<template>
  <div class="bonuses-table">
    <BContainer>
      <div class="bonuses-table__content">
        <BTable :fields="fields" :items="items" />
      </div>
    </BContainer>
  </div>
</template>

<script>
export default {
  props: {
    items: {
      type: Array,
      required: true
    }
  },
  data() {
    return {
      fields: [
        { key: 'amount', label: this.$t('bonusesTable.amountTitle') },
        { key: 'percent', label: this.$t('bonusesTable.percentTitle') }
      ]
    }
  }
}
</script>

<style lang="scss">
.bonuses-table {
  margin-bottom: 39px;

  &__content {
    padding-left: 17px;
  }

  table th,
  table td {
    font-size: 12px;
    font-weight: 400;
    line-height: 15px;
    vertical-align: middle;
    border: none;
    border-bottom: 1px solid $gray;
  }

  table th {
    padding-top: 8px;
    padding-bottom: 13px;
  }

  table td {
    padding: 18px 0;
  }

  table tr td:first-of-type {
    width: 60%;
    color: $text-color;
  }

  table tr td:last-of-type {
    font-weight: 700;
  }

  @include media-breakpoint-up(md) {
    &__content {
      max-width: 405px;
      padding-left: 0;
    }

    table tr td:first-of-type {
      width: 65%;
    }
  }

  @include media-breakpoint-up(lg) {
    &__content {
      max-width: 425px;
      margin-left: 80px;
    }
  }

  @include media-breakpoint-up(xl) {
    margin-bottom: 57px;

    &__content {
      max-width: 515px;
      margin-left: 100px;
    }

    table tr td:first-of-type {
      width: 70%;
    }

    table th,
    table td {
      font-size: 13px;
      line-height: 16px;
    }
  }

  @include media-breakpoint-up(xxl) {
    &__content {
      max-width: 480px;
      margin-left: 118px;
    }

    table tr td:last-of-type {
      font-size: 14px;
      line-height: 18px;
    }

    table td {
      padding: 17px 0;
    }
  }
}
</style>
