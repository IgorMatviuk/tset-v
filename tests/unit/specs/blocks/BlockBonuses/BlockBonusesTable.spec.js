import { shallowMount, createLocalVue } from '@vue/test-utils'
import _cloneDeep from 'lodash.clonedeep'

import { BContainer, BTable } from 'bootstrap-vue'

import BlockBonusesTable from '@/components/blocks/BlockBonuses/BlockBonusesTable'

const localVue = createLocalVue()
localVue.component('BContainer', BContainer)
localVue.component('BTable', BTable)

describe('BlockBonusesTable', () => {
  let items

  function createBlockBonusesTable() {
    const wrapper = shallowMount(BlockBonusesTable, {
      localVue,
      propsData: {
        items: _cloneDeep(items)
      },
      mocks: {
        $t: jest.fn()
      },
      stubs: ['svg-icon']
    })

    return wrapper
  }

  beforeEach(() => {
    items = [
      {
        amount: '',
        percent: ''
      },
      {
        amount: '',
        percent: ''
      },
      {
        amount: '',
        percent: ''
      }
    ]
  })

  test('create bonuses vip block', () => {
    const wrapper = createBlockBonusesTable()

    expect(wrapper.element).toMatchSnapshot()
  })
})
