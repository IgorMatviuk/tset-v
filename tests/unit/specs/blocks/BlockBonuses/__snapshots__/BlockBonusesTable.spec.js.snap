// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockBonusesTable create bonuses vip block 1`] = `
<div
  class="bonuses-table"
>
  <bcontainer-stub
    tag="div"
  >
    <div
      class="bonuses-table__content"
    >
      <btable-stub
        apiurl=""
        currentpage="1"
        emptyfilteredtext="There are no records matching your request"
        emptytext="There are no records to show"
        fields="[object Object],[object Object]"
        filterdebounce="0"
        items="[object Object],[object Object],[object Object]"
        labelsortasc="Click to sort Ascending"
        labelsortclear="Click to clear sorting"
        labelsortdesc="Click to sort Descending"
        perpage="0"
        selectedvariant="active"
        selectmode="multi"
        sortby=""
        sortcompareoptions="[object Object]"
        sortdirection="asc"
        value=""
      />
    </div>
  </bcontainer-stub>
</div>
`;
